### PR TITLE
#1618: Duplicated items in Used in Entities filters

### LINKED ISSUE #1618: Duplicated items in Used in Entities filters
### Description
The issue is not reproducible constantly, but I have seen it on all **Used in** filters.
Usually, after the web page is refreshed, the issue disappears 
I will continue to investigate it and will add additional info if I find any
A low Severity is added

### Precondition (*)
1. Install Magento with Adobe Stock Integration
2. Enhanced Media Gallery is Enabled
3. A saved image Preview from Adobe Stock

### Steps to reproduce (*)
1. Go to **Content - Media Gallery**
2. Click **Filters**
3. Expand, for example, **Used in Pages** filter
4. Start typing _page_ into **Page Title** field

### Actual result (*)
The found entities are duplicated
duplicated
duplicated2

### PR DESCRIPTION
**Pull request by joweecaquicla**
### Description (*)

Prevent simultaneous requests when user types search key in `ui-select`.
### Related Pull Requests

_N/A_
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/adobe-stock-integration/issues/1618

### Manual testing scenarios (*)

1. Go to Content - Media Gallery
2. Click Filters
3. Expand, for example, Used in Pages filter
4. Start typing page into Page Title field

### Questions or comments

_N/A_
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by joweecaquicla** [comment]
@magento run all tests

File: app/code/Magento/Ui/view/base/web/js/form/element/ui-select.js [hunk 1]
```diff
@@ -153,8 +153,8 @@ define([
             showPath: true,
             labelsDecoration: false,
             disableLabel: false,
-            filterRateLimit: 500,
-            filterRateLimitMethod: 'notifyAtFixedRate',
+            filterRateLimit: 1000,
+            filterRateLimitMethod: 'notifyWhenChangesStop',
```
Comment (sivaschenko) [inline app/code/Magento/Ui/view/base/web/js/form/element/ui-select.js]: I don't think these defaults changes are necessary, also this is backward incompatible change

File: app/code/Magento/Ui/view/base/web/js/form/element/ui-select.js [hunk 2]
```diff
@@ -1167,7 +1167,7 @@ define([
                 return;
             }
 
-            if (searchKey !== this.lastSearchKey) {
+            if (searchKey !== this.lastSearchKey || currentPage === 1) {
```
Comment (sivaschenko) [inline app/code/Magento/Ui/view/base/web/js/form/element/ui-select.js]: to be honest I think we can even simplify this condition (considering how currentPage is calculated at the beginning of this method)
```suggestion
            if (currentPage === 1) {
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @joweecaquicla ! Please see my comments

**Comment by joweecaquicla** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@chalov-anton please test together with https://github.com/magento/adobe-stock-integration/pull/1655

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7902](https://jira.corp.magento.com/browse/ENGCOM-7902) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by sivaschenko** [comment]
ui-select functionality is covered by various MFTF tests

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

Tested together with Adobe Stock PR - https://github.com/magento/adobe-stock-integration/pull/1655
The original issue is not reproducible.

Added a scenario to regression scope - https://studio.cucumber.io/projects/131313/test-plan/folders/1320712/scenarios/5021347

**Comment by sivaschenko** [comment]
@magento run Functional Tests CE,Integration Tests

**Pull request merged**